dask/dataframe/core.py

```diff
@@ -3411,18 +3412,21 @@ def astype(self, dtype):
         # categorical dtypes. This operation isn't allowed currently anyway. We
         # get the metadata with a non-empty frame to throw the error instead of
         # segfaulting.
-        if is_dataframe_like(self._meta) and is_categorical_dtype(dtype):
+        if is_dataframe_like(self._meta) and isinstance(dtype, pd.CategoricalDtype):
             meta = self._meta_nonempty.astype(dtype)
         else:
             meta = self._meta.astype(dtype)
         if hasattr(dtype, "items"):
             set_unknown = [
                 k
                 for k, v in dtype.items()
-                if is_categorical_dtype(v) and getattr(v, "categories", None) is None
+                if (isinstance(pd.api.types.pandas_dtype(v), pd.CategoricalDtype))
```

Comment:
Why do we need `pd.api.types.pandas_dtype` here?

Reply:
We can get „category“ as string instead of a dtype object here,pandas_dtype takes care of the proper conversion

Reply:
Ah, I see. Thanks for clarifying 👍

Reply:
This would work with string or dtype, right?

```python
if pd.api.types.is_dtype_equal(v, pd.CategoricalDtype())
```

Reply:
Yes, generally I prefer ``pandas_dtype`` though since it raises if you provide something invalid